### DISCUSSION TITLE
Install this onto a Mac

### DISCUSSION THREAD

**Discussion by tommytraxx** [keirf/greaseweazle#434] [Q&A]
Hiya. I have a bunch of sample floppys from an EMU SP1200 drum machine from the 90's. I've had an SDD drive installed in the unit and now need to transfer the floppies onto a USB drive via the Grease Weasel. The guy that did it only knows the PC way but I use a Mac. I'm not a coder so trying to get my head around this. Please help!

**Comment by keirf** [comment]
Well it would be useful to know his PC way, and then adjust it for Mac.

But to install on Mac you will need to get acquainted with Terminal, and use the command line. The step-by-step setup instructions are [here](https://github.com/keirf/greaseweazle/wiki/Software-Installation#macos). 

Once you have the tools installed and your drive set up, you may be able to generate HFE image files as easily as (creating a new file called `name.hfe`):
```
gw read name.hfe::bitrate=250
```

**Reply by tommytraxx** [reply]
Thanks so much for coming back to me. I'm aquatinted with Terminal and I've followed those instructions. Installing Python was fine but installing Pipx was where I came unstuck. This is what I got:

You have not agreed to the Xcode license agreements, please run 'sudo xcodebuild -license' from within a Terminal window to review and agree to the Xcode license agreements.
WARNING: Discarding git+https://github.com/keirf/greaseweazle@latest. Command errored out with exit status 69: git clone -q https://github.com/keirf/greaseweazle /private/var/folders/8f/vcjmq4953ks_7jdng3vc2lp40000gp/T/pip-req-build-dt0wni3n Check the logs for full command output.
ERROR: Command errored out with exit status 69: git clone -q https://github.com/keirf/greaseweazle /private/var/folders/8f/vcjmq4953ks_7jdng3vc2lp40000gp/T/pip-req-build-dt0wni3n Check the logs for full command output.
WARNING: You are using pip version 21.2.3; however, version 24.0 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.

**Reply by keirf** [reply]
Which specific command gave that error?

**Reply by keirf** [reply]
Actually, look at your first line of error message:
```
You have not agreed to the Xcode license agreements, please run 'sudo xcodebuild -license' from within a Terminal window to review and agree to the Xcode license agreements.
```
I haven't seen that message before, but you should probably do that and then try again. That is, you should run this in Terminal and enter your password if prompted:
```
sudo xcodebuild -license
```

**Reply by tommytraxx** [reply]
brew install pipx
pipx ensurepath
pipx install git+https://github.com/keirf/greaseweazle@latest

**Reply by keirf** [reply]
Right. So now you need to agree the Xcode license and then retry the last pipx line:
```
sudo xcodebuild -license
pipx install git+https://github.com/keirf/greaseweazle@latest
```
